blues/moves.py

```diff
@@ -177,70 +178,304 @@ def move(self, context):
-        Parameters
-        ----------
-        moves : blues.ncmc.Model object or list of n blues.ncmc.Model-like objects
-            Specifies the possible moves to be performed.
+class SmartDartMove(RandomLigandRotationMove):
+    def __init__(self, structure, basis_particles, dart_size=0.2*unit.nanometers, resname='LIG'):
+        super(SmartDartMove, self).__init__(structure, resname=resname)
+        self.dartboard = []
+        self.n_dartboard = []
+        self.particle_pairs = []
+        self.particle_weights = []
+        self.basis_particles = basis_particles
+        self.dart_size = dart_size
+        self.calculateProperties()
+
+    def dartsFromParmEd(self, coord_files, topology=None):
+        """
+        Used to setup darts from a generic coordinate file, through MDtraj using the basis_particles to define
+        new basis vectors, which allows dart centers to remain consistant through a simulation.
+        This adds to the self.n_dartboard, which defines the centers used for smart darting.
+        Arguments
+        ---------
+        system: simtk.openmm.system
+            Openmm System corresponding to the system to smart dart.
+        coord_files: list of str
+            List containing coordinate files of the system for smart darting.
+        basis_particles: list of 3 ints
+            Specifies the 3 indices of particles whose coordinates will be used
+            as basis vectors. If None is specified, uses those found in basis particles.
+            If None uses self.basis_particles instead.
+        Returns
+        -------
+        n_dartboard: list of 1x3 np.arrays
+            Center of mass coordinates of atom_indices particles in new basis set.
+        """
+        n_dartboard = []
+        dartboard = []
+        for coord_file in coord_files:
+            if topology == None:
+                temp_md = parmed.load_file(coord_file)
+            else:
+                temp_md = parmed.load_file(topology, xyz=coord_file)
+            context_pos = temp_md.positions.in_units_of(unit.nanometers)
+            lig_pos = np.asarray(context_pos._value)[self.atom_indices]*unit.nanometers
+            particle_pos = np.asarray(context_pos._value)[self.basis_particles]*unit.nanometers
+            self.calculateProperties()
+            self.center_of_mass = self.getCenterOfMass(lig_pos, self.masses)
+            #get particle positions
+            new_coord = self._findNewCoord(particle_pos[0], particle_pos[1], particle_pos[2], self.center_of_mass)
+            #keep this in for now to check code is correct
+            #old_coord should be equal to com
+            old_coord = self._findOldCoord(particle_pos[0], particle_pos[1], particle_pos[2], new_coord)
+            n_dartboard.append(new_coord)
+            dartboard.append(old_coord)
+
+        self.n_dartboard = n_dartboard
+        self.dartboard = dartboard
+
+    def dartsFromMDTraj(self, system, file_list, topology=None):
```

Comment:
Seems like there is substantial duplication between `dartsFromMDTraj` and `dartsFromParmEd`, e.g. most of my doc string comments from above apply here, and also the code seems fairly similar except for how you load files and certain other details. Should you reduce code duplication?